lib/internal/http2/compat.js

```diff
@@ -552,7 +552,7 @@ class Http2ServerResponse extends Stream {
   }
 
   getHeaders() {
-    return { ...this[kHeaders] };
```

Comment:
The code change does not align with the changed title. Should this be:

```js
const headers = Object.create(null);
Object.assign(headers, this[kHeaders]);
return headers;
```

Reply:
`this[kHeaders]` was initialized originally (`L448` below) using `Object.create(null)` so I think this is okay, right?

https://github.com/nodejs/node/blob/53eb264cb16f5c1815d4ccfda585b462f9fac92a/lib/internal/http2/compat.js#L437-L449

Reply:
With this change it would be possible to change internal state by manipulating the exported object. That's probably not intentional. We should therefore export a new object.